src/main/java/com/security/contests/repository/CustomDAOImpl.java

```diff
@@ -46,6 +46,34 @@ public int saveUserRoleData(Long userId, Long roleId) {
 		return query.executeUpdate();
 	}
 
+	@Override
+	public int saveWalletData(Long userId, Long balance) {
+		final String sql = "INSERT INTO wallet(balance,user_id) values(?,?)";
+		Query query = em.createNativeQuery(sql);
+		query.setParameter(2, userId);
+		query.setParameter(1, balance);
+		return query.executeUpdate();
```

Comment:
This is failing with Illegal state and expecting UserRoles from user to be saved first. We need to bypass that check as it's already saved.